### PR TITLE
Mainnet WETH Deployment

### PR DESCRIPTION
**Pull request by jflatow**
~~WIP~~ Ready for review now.

Changes made in preparation for launching a ~~mainnet/ETH~~`mainnet/WETH` deployment.

- adds `mainnet-weth` scenario base
- adds configuration for `mainnet/weth` deploy with `wstETH` and `cbETH` as collateral assets
- adds a `WstETHPriceFeed.sol` contract that calculates the `wstETH / USD` price using the `stETH / USD` price feed and the value of `wstETH:stETH`
- updates scenarios to work with ETH-based markets, where possible
- adds a `matchesDeployment` filter to allow for scenarios that specifically target (or avoid) the `mainnet-weth` deploy
- adds a `getRewardToken` fn to context, to allow retrieving the reward token even when it is not a collateral asset

Remaining todos before we can launch:

- [x] set the supply caps back to 0
- [x] replace the `cbETH` price feed with the actual Chainlink `cbETH / USD` price feed (when it's available)
- [x] decide on actual configuration params
- [x] get OZ audit on `WstETHPriceFeed.sol`
- [x] launch a new version of the Liquidator Bot for `mainnet/eth`
- [x] deploy new Bulker (#611)

### CODE REVIEW

File: deployments/mainnet/eth/configuration.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    "borrowBase": 0.015
+  },
+  "tracking": {
+    "indexScale": "1e15",
+    "baseSupplySpeed": "0e15",
+    "baseBorrowSpeed": "0e15",
+    "baseMinForRewards": "1000000e6"
+  },
+  "rewardTokenAddress": "0xc00e94cb662c3520282e6f5717214004a7f26888",
+  "assets": {
```
Comment (kevincheng96) [inline deployments/mainnet/eth/configuration.json:29]: Thoughts on also supporting a stablecoin like USDC to better facilitate the shorting use-case?

Comment (kevincheng96) [reply deployments/mainnet/eth/configuration.json:29]: Since with the current configuration, the only use-case is levering up on staked ETH.

File: .github/workflows/deploy-market.yaml [hunk 2]
```diff
@@ -32,7 +33,7 @@ jobs:
       - name: Seacrest
         uses: hayesgm/seacrest@v1
         with:
-          ethereum_url: "${{ fromJSON('{\"fuji\":\"https://api.avax-test.network/ext/bc/C/rpc\",\"kovan\":\"https://kovan.infura.io/v3/$INFURA_KEY\",\"mainnet\":\"https://mainnet.infura.io/v3/$INFURA_KEY\",\"goerli\":\"https://goerli.infura.io/v3/$INFURA_KEY\",\"mumbai\":\"https://polygon-mumbai.infura.io/v3/$INFURA_KEY\"}')[inputs.network] }}"
+          ethereum_url: "${{ fromJSON('{\"fuji\":\"https://api.avax-test.network/ext/bc/C/rpc\",\"kovan\":\"https://kovan.infura.io/v3/$INFURA_KEY\",\"mainnet\":\"https://mainnet.infura.io/v3/$INFURA_KEY\",\"mainnet-eth\":\"https://mainnet.infura.io/v3/$INFURA_KEY\",\"goerli\":\"https://goerli.infura.io/v3/$INFURA_KEY\",\"mumbai\":\"https://polygon-mumbai.infura.io/v3/$INFURA_KEY\"}')[inputs.network] }}"
```
Comment (scott-silver) [inline .github/workflows/deploy-market.yaml:36]: added the `mainnet-eth` deployment to the Github Actions.

for the `deploy-market` action, `mainnet` uses `https://mainnet.infura.io/v3/$INFURA_KEY`, but for `prepare-migration` and `enact-migration` it uses `https://mainnet-eth.compound.finance\`

wondering if there is a reason for the difference? or should we pick one and use them across the actions

File: deployments/mainnet/eth/configuration.json (hunk 1 above)
Comment (scott-silver) [reply deployments/mainnet/eth/configuration.json:29]: Sounds like we're sticking with `cbETH` and `wstETH` for the launch

File: .github/workflows/deploy-market.yaml (hunk 2 above)
Comment (scott-silver) [reply .github/workflows/deploy-market.yaml:36]: updated

File: deployments/mainnet/eth/configuration.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    "cbETH": {
+      "address": "0xBe9895146f7AF43049ca1c1AE358B0541Ea49704",
+      "priceFeed": "0xCfE54B5cD566aB89272946F602D76Ea879CAb4a8",
+      "decimals": "18",
+      "borrowCF": 0.80,
+      "liquidateCF": 0.85,
+      "liquidationFactor": 0.95,
+      "supplyCap": "500e18"
```
Comment (scott-silver) [inline deployments/mainnet/eth/configuration.json:37]: bumped the supply caps to get scenarios running

- [ ] reset supplyCaps to 0

File: .github/workflows/deploy-market.yaml [hunk 4]
```diff
@@ -32,7 +33,7 @@ jobs:
       - name: Seacrest
         uses: hayesgm/seacrest@v1
         with:
-          ethereum_url: "${{ fromJSON('{\"fuji\":\"https://api.avax-test.network/ext/bc/C/rpc\",\"kovan\":\"https://kovan.infura.io/v3/$INFURA_KEY\",\"mainnet\":\"https://mainnet.infura.io/v3/$INFURA_KEY\",\"goerli\":\"https://goerli.infura.io/v3/$INFURA_KEY\",\"mumbai\":\"https://polygon-mumbai.infura.io/v3/$INFURA_KEY\"}')[inputs.network] }}"
```
Comment (Cumop21) [inline .github/workflows/deploy-market.yaml:35]: jflatow/mainnet-eth

**Review by Cumop21 — APPROVED** [review]

File: scenario/SupplyScenario.ts [hunk 5]
```diff
@@ -250,7 +250,7 @@ scenario(
         asset: baseAsset.address,
         amount: 100n * scale,
       })
-    ).to.be.revertedWith('ERC20: transfer amount exceeds allowance');
+    ).to.be.reverted;
```
Comment (scott-silver) [inline scenario/SupplyScenario.ts:253]: `WETH` reverts without reason strings, unfortunately ☹️

File: scenario/RewardsScenario.ts [hunk 6]
```diff
@@ -156,8 +156,8 @@ scenario(
     const baseScale = (await comet.baseScale()).toBigInt();
     const toBorrow = 1_000n * baseScale;
 
-    const [rewardTokenAddress, rescaleFactor] = await rewards.rewardConfig(comet.address);
-    const rewardToken = context.getAssetByAddress(rewardTokenAddress);
+    const [_, rescaleFactor] = await rewards.rewardConfig(comet.address);
+    const rewardToken = await context.getRewardToken();
```
Comment (scott-silver) [inline scenario/RewardsScenario.ts:160]: we could have `context.getRewardToken` return both the `rewardToken` instance and the `rescaleFactor`, which would eliminate the need to call `rewards.rewardConfig` on line 159

Comment (jflatow) [reply scenario/RewardsScenario.ts:160]: `getRewardConfig`?

File: scenario/SupplyScenario.ts (hunk 5 above)
Comment (jflatow) [reply scenario/SupplyScenario.ts:253]: This kinda sucks, another case where I think we should at least preserve the error information in a comment in case we find a way to restore strength later

File: .github/workflows/deploy-market.yaml [hunk 7]
```diff
@@ -9,6 +9,7 @@ on:
           - fuji
           - kovan
           - mainnet
+          - mainnet-eth
```
Comment (jflatow) [inline .github/workflows/deploy-market.yaml:12]: Can we get rid of this now?

File: deployments/mainnet/eth/deploy.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import { Deployed, DeploymentManager } from '../../../plugins/deployment_manager';
+import { DeploySpec, deployComet } from '../../../src/deploy';
+import { getConfiguration, NetworkConfiguration } from '../../../src/deploy/NetworkConfiguration';
+
+export default async function deploy(deploymentManager: DeploymentManager, deploySpec: DeploySpec): Promise<Deployed> {
+  // XXX configuration params for assets are just placeholders
```
Comment (jflatow) [inline deployments/mainnet/eth/deploy.ts:6]: We should ask Gauntlet to start thinking about params too

File: deployments/mainnet/eth/deploy.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  // Deploy all Comet-related contracts
+  const deployed = await deployComet(
+    deploymentManager,
+    deploySpec,
+    { assetConfigs }
+  );
+  const { comet } = deployed;
+
+  // Get a handle to the WETH contract for bulker
+  const WETH = await deploymentManager.existing('WETH', await comet.baseToken());
```
Comment (jflatow) [inline deployments/mainnet/eth/deploy.ts:35]: Is it not in spider yet? (e.g. `deploymentManager.contract('WETH')`)

File: deployments/mainnet/eth/deploy.ts [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  let { assetConfigs } = await getConfiguration(deploymentManager);
+  assetConfigs = (assetConfigs || []).map(ac => {
+    return ac.asset === wstETHaddress ? { ...ac, ...{ priceFeed: wstETHPriceFeed.address } } : ac;
+  });
+
+  // Deploy all Comet-related contracts
+  const deployed = await deployComet(
+    deploymentManager,
+    deploySpec,
+    { assetConfigs }
```
Comment (jflatow) [inline deployments/mainnet/eth/deploy.ts:30]: Something seems off here, we shouldn't have to do all this... 🤔

**Review by jflatow — COMMENTED** [review]
Looking great, couple comments/questions 💪

File: deployments/mainnet/eth/deploy.ts (hunk 10 above)
Comment (scott-silver) [reply deployments/mainnet/eth/deploy.ts:30]: The additional setup is so that we can deploy the `wstETHPriceFeed` contract and then pass it in as an argument to `deployComet`, overriding the value from `configuration.json`.

We normally don't have to do this since all of our price feeds are already deployed.

One other option would be to deploy the price feed manually in advance, and just add that address to the configuration file.

File: deployments/mainnet/eth/deploy.ts (hunk 8 above)
Comment (scott-silver) [reply deployments/mainnet/eth/deploy.ts:6]: Started a conversation in the Gauntlet channel

File: .github/workflows/deploy-market.yaml (hunk 7 above)
Comment (scott-silver) [reply .github/workflows/deploy-market.yaml]: yep, removed

File: deployments/mainnet/eth/deploy.ts (hunk 9 above)
Comment (scott-silver) [reply deployments/mainnet/eth/deploy.ts]: we don't even need to do that. we can just pass in `await comet.baseToken()` to the Bulker constructor. updated

File: scenario/SupplyScenario.ts (hunk 5 above)
Comment (scott-silver) [reply scenario/SupplyScenario.ts:253]: sure, added the error messages back as comments.

as long as we're running these scenarios against a WETH-based deployment we'll have to stick with `.reverted`, though, since the WETH contract we're using isn't ever going to return revert strings.

Comment (jflatow) [reply scenario/SupplyScenario.ts:253]: maybe one day we'll have network specific revert reasons 🤷

File: deployments/mainnet/eth/deploy.ts [hunk 11]
```diff
@@ -0,0 +1,45 @@
+import { Deployed, DeploymentManager } from '../../../plugins/deployment_manager';
+import { DeploySpec, deployComet } from '../../../src/deploy';
+import { getConfiguration, NetworkConfiguration } from '../../../src/deploy/NetworkConfiguration';
+
+export default async function deploy(deploymentManager: DeploymentManager, deploySpec: DeploySpec): Promise<Deployed> {
+  // XXX configuration params for assets are just placeholders
+
+  // Deploy WstETHPriceFeed
+  const wstETHPriceFeed = await deploymentManager.deploy(
+    'wstETHPriceFeed',
```
Comment (jflatow) [inline deployments/mainnet/eth/deploy.ts:8]: The alias should be the same as what spider gets, which I think should be `wstETH:priceFeed`

File: scenario/SupplyScenario.ts (hunk 5 above)
Comment (scott-silver) [reply scenario/SupplyScenario.ts:253]: oh, that's an interesting idea.

we could write a wrapper function that the function to execute and a map of networks to the expected error message. for anything

File: deployments/mainnet/eth/deploy.ts (hunk 10 above)
Comment (jflatow) [reply deployments/mainnet/eth/deploy.ts:28]: I think we might be able to get rid of everything except the deploy if we use ```getContractAddress(`${assetName}:priceFeed`, assetConfig.priceFeed)``` here https://github.com/compound-finance/comet/blob/main/src/deploy/NetworkConfiguration.ts#L98 i.e. with the current one as the fallback (and I guess here too for parity https://github.com/compound-finance/comet/blob/main/src/deploy/NetworkConfiguration.ts#L134)

File: scenario/RewardsScenario.ts (hunk 6 above)
Comment (scott-silver) [reply scenario/RewardsScenario.ts:160]: i like it. added a `getRewardConfig` function context.

File: deployments/mainnet/eth/deploy.ts [hunk 12]
```diff
@@ -5,28 +5,16 @@ import { getConfiguration, NetworkConfiguration } from '../../../src/deploy/Netw
 export default async function deploy(deploymentManager: DeploymentManager, deploySpec: DeploySpec): Promise<Deployed> {
   // Deploy WstETHPriceFeed
   const wstETHPriceFeed = await deploymentManager.deploy(
-    'wstETHPriceFeed',
+    'wstETH:priceFeed',
```
Comment (jflatow) [inline deployments/mainnet/eth/deploy.ts:8]: 🎉

File: deployments/mainnet/eth/deploy.ts (hunk 10 above)
Comment (scott-silver) [reply deployments/mainnet/eth/deploy.ts]: that's awesome. made the update

File: deployments/mainnet/eth/configuration.json [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+    "indexScale": "1e15",
+    "baseSupplySpeed": "0e15",
+    "baseBorrowSpeed": "0e15",
+    "baseMinForRewards": "1000000e6"
+  },
+  "rewardTokenAddress": "0xc00e94cb662c3520282e6f5717214004a7f26888",
+  "assets": {
+    "cbETH": {
+      "address": "0xBe9895146f7AF43049ca1c1AE358B0541Ea49704",
+      "priceFeed": "0xCfE54B5cD566aB89272946F602D76Ea879CAb4a8",
```
Comment (jflatow) [inline deployments/mainnet/eth/configuration.json:32]: XXX change to upcoming cbETH/USD feed

**Comment by jflatow** [comment]
For setting the supply caps from 0, we need a separate migration w/ a proposal. For USDC we also seeded reserves, not sure if we want to try to do that here. Doing rewards after I think makes sense

File: deployments/mainnet/eth/configuration.json (hunk 13 above)
Comment (jflatow) [reply deployments/mainnet/eth/configuration.json]: Updated to https://etherscan.io/address/0x67eF3CAF8BeB93149F48e8d20920BEC9b4320510

File: scenario/BulkerScenario.ts [hunk 14]
```diff
@@ -168,3 +169,94 @@ scenario(
...
+
+    // Final expectations
+    const baseIndexScale = (await comet.baseIndexScale()).toBigInt();
+    const baseSupplyIndex = (await comet.totalsBasic()).baseSupplyIndex.toBigInt();
+    const baseTransferred = getExpectedBaseBalance(toTransferBase, baseIndexScale, baseSupplyIndex);
+    expect(await comet.collateralBalanceOf(albert.address, collateralAsset.address)).to.be.equal(toSupplyCollateral);
+    expect(await baseAsset.balanceOf(albert.address)).to.be.equal(toBorrowBase);
+    expect(await comet.balanceOf(betty.address)).to.be.equal(baseTransferred);
+    // NOTE: differs from the equivalent scenario for non-ETH markets
+    expect(await comet.borrowBalanceOf(albert.address)).to.be.equal(toBorrowBase + toTransferBase - (toSupplyEth - toWithdrawEth));
```
[the remark this answers is not in the document]
Comment (jflatow) [reply scenario/BulkerScenario.ts:329]: Just to make sure I understand, if the scenario changed the collateral asset from ETH to something else it would be the same right?

**Comment by jflatow** [comment]
- [ ] We also need to make the name consistent everywhere (cWETHv3)

File: scenario/BulkerScenario.ts (hunk 14 above)
[the remark this answers is not in the document]
Comment (scott-silver) [reply scenario/BulkerScenario.ts:329]: The `non-WETH` version of this test has these two assertions instead:

```
expect(await comet.borrowBalanceOf(albert.address)).to.be.equal(toBorrowBase + toTransferBase);
expect(await comet.collateralBalanceOf(albert.address, WETH.address)).to.be.equal(toSupplyEth - toWithdrawEth);
```

So in a USDC deployment, the borrow balance is how much the user borrowed and transferred. As well, the user has a collateral balance of WETH equal to how much ETH they supplied minus how much they borrowed.

But in this case, since the base is WETH, then the borrow balance is how much WETH was borrowed, how much WETH was transferred, how much ETH was borrowed (minus how much ETH was supplied)

File: scenario/GovernanceScenario.ts [hunk 15]
```diff
@@ -143,7 +143,68 @@ scenario('add new asset',
...
+    const borrowAmount = 1000n * (await comet.baseScale()).toBigInt();
+    await dogecoin.connect(albert.signer).approve(comet.address, exp(100, 8));
+    await albert.supplyAsset({ asset: dogecoin.address, amount: exp(100, 8) });
+    await albert.withdrawAsset({ asset: baseAssetAddress, amount: borrowAmount });
+
+    expect(await albert.getCometCollateralBalance(dogecoin.address)).to.be.equal(exp(100, 8));
+    expect(await albert.getCometBaseBalance()).to.be.equal(-borrowAmount);
+  });
+
+scenario('add new asset (mainnet-weth)',
```
Comment (kevincheng96) [inline scenario/GovernanceScenario.ts:155]: Curious why we need to separate out this scenario? Shouldn't it just work for both deployments?

File: scenario/RewardsScenario.ts [hunk 16]
```diff
@@ -21,7 +21,7 @@ function calculateRewardsOwed(
 scenario(
   'Comet#rewards > can claim supply rewards for self',
   {
-    filter: async (ctx) => await isRewardSupported(ctx),
+    filter: async (ctx) => await isRewardSupported(ctx) && !matchesDeployment(ctx, [{network: 'mainnet', deployment: 'weth'}]),
```
Comment (kevincheng96) [inline scenario/RewardsScenario.ts:24]: We should add an XXX to remove this filter once rewards are supported on the WETH deployment.

File: test/wsteth-price-feed.ts [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+  await wstETHPriceFeed.deployed();
+
+  return {
+    simpleWstETH,
+    stETHPriceFeed,
+    wstETHPriceFeed
+  };
+}
+
+const testCases = [
```
Comment (kevincheng96) [inline test/wsteth-price-feed.ts:29]: Weird edge case to maybe test is if the price returned is negative (Chainlink returns `int256` for price). I don't think it really matters since Comet will just revert if it's a negative price, but let's make sure we do return a negative price back.

**Review by kevincheng96 — COMMENTED** [review]
Looks great so far!

File: scenario/GovernanceScenario.ts (hunk 15 above)
Comment (scott-silver) [reply scenario/GovernanceScenario.ts:155]: the original test attempts to borrow 1,000 units of the base asset using 100 assets of DOGE (with a price of $1,000 per DOGE).

that works with a USDC base, since a $1,000 borrow can be collateralized with $100,000 worth of DOGE.

but borrowing 1,000 ETH costs about $1.5M, so the borrow is uncollateralized

File: scenario/RewardsScenario.ts (hunk 16 above)
Comment (scott-silver) [reply scenario/RewardsScenario.ts:24]: i guess alternatively, we could delete `rewardTokenAddress` from the `mainnet/WETH` configuration so that the `isRewardSupported` check works as intended

File: contracts/WstETHPriceFeed.sol [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+    function latestRoundData() override external view returns (
+        uint80 roundId,
+        int256 answer,
+        uint256 startedAt,
+        uint256 updatedAt,
+        uint80 answeredInRound
+    ) {
+        (uint80 roundId, int256 stETHPrice, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound) = AggregatorV3Interface(stETHtoUSDPriceFeed).latestRoundData();
+        uint256 tokensPerStEth = IWstETH(wstETH).tokensPerStEth();
+        int price = stETHPrice * signed256(wstETHScale) / signed256(tokensPerStEth);
```
[the remark this answers is not in the document]
Comment (scott-silver) [reply contracts/WstETHPriceFeed.sol]: yep, that works too. updated

File: scenario/RewardsScenario.ts (hunk 16 above)
Comment (scott-silver) [reply scenario/RewardsScenario.ts]: deleted `rewardTokenAddress` from the configuration file, and deleted this filter

File: test/wsteth-price-feed.ts (hunk 17 above)
Comment (scott-silver) [reply test/wsteth-price-feed.ts:29]: good idea. added a negative price to the test cases

File: scenario/GovernanceScenario.ts (hunk 15 above)
Comment (kevincheng96) [reply scenario/GovernanceScenario.ts:155]: Ah I see, that is indeed annoying. Another approach could be to set the borrow amount to be the `borrowBaseMin` instead of a constant. But this works too.

**Review by kevincheng96 — COMMENTED** [review]
Cool, I think everything looks good now. Just waiting on params to be finalized

File: contracts/WstETHPriceFeed.sol [hunk 19]
```diff
@@ -70,7 +70,7 @@ contract WstETHPriceFeed is AggregatorV3Interface {
     ) {
         (uint80 roundId, int256 stETHPrice, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound) = AggregatorV3Interface(stETHtoUSDPriceFeed).latestRoundData();
         uint256 tokensPerStEth = IWstETH(wstETH).tokensPerStEth();
-        int price = stETHPrice * signed256(wstETHScale) / signed256(tokensPerStEth);
+        int256 price = stETHPrice * int256(wstETHScale) / signed256(tokensPerStEth);
```
Comment (jflatow) [inline contracts/WstETHPriceFeed.sol:73]: Since we are assuming `wstETHScale` is already guaranteed to be < int256.max?

Comment (scott-silver) [reply contracts/WstETHPriceFeed.sol:73]: yeah, that's the assumption:

https://github.com/compound-finance/comet/pull/587#discussion_r1009744397

File: scenario/constraints/SupplyCapConstraint.ts [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+    const supplyCaps = requirements.supplyCaps;
+    if (supplyCaps !== undefined) {
+      const solutions = [];
+      solutions.push(async function barelyMeet(context: T) {
+        const supplyAmountPerAsset = {};
+        for (const [assetName, rawAmount] of Object.entries(supplyCaps)) {
+          const asset = await getAssetFromName(assetName, context);
+          const decimals = await asset.token.decimals();
+          const amount = parseAmount(rawAmount);
+          expect(amount.op).to.equal(ComparisonOp.GTE, `Operation ${amount.op} not supported (yet) by supply cap constraint`);
```
Comment (kevincheng96) [inline scenario/constraints/SupplyCapConstraint.ts:19]: What does this line do?

File: contracts/WstETHPriceFeed.sol (hunk 19 above)
Comment (jflatow) [reply contracts/WstETHPriceFeed.sol:73]: Yep 👍 comet has `10**18` max constant and changing that would be a massive breaking change

File: scenario/GovernanceScenario.ts (hunk 15 above)
Comment (jflatow) [reply scenario/GovernanceScenario.ts:155]: Shouldn't we just have a PriceConstraint?

Comment (scott-silver) [reply scenario/GovernanceScenario.ts:155]: that's a clever idea. So you'd specify a price for the base asset and/or the collateral assets

we're having a similar conversation on the [Polygon PR](https://github.com/compound-finance/comet/pull/598#discussion_r1013496694).

Another approach would be to write a helper function returned the amount of a given asset required to cover a borrow of X base units

Comment (jflatow) [reply scenario/GovernanceScenario.ts:155]: Haha, yep just commented there, we can pick one ;) But I think its more desirable to use literal numbers in properties where we can (👨‍🍳 has long advocated for this and I kind of agree)

Comment (scott-silver) [reply scenario/GovernanceScenario.ts:155]: #615

File: deployments/mainnet/weth/configuration.json [hunk 21]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "Compound WETH",
+  "symbol": "cWETHv3",
+  "baseToken": "WETH",
+  "baseTokenAddress": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
+  "baseTokenPriceFeed": "0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419",
+  "borrowMin": "100e6",
```
[the remark this answers is not in the document]
Comment (scott-silver) [reply deployments/mainnet/weth/configuration.json:6]: bumping this so we don't forget: `borrowMin`, `targetReserves`, and `baseMinForRewards` are all 6 decimal values, not 18 decimal values

[the remark this answers is not in the document]
Comment (kevincheng96) [reply deployments/mainnet/weth/configuration.json:6]: Here's what I'm thinking for the tracking index values:
```
trackingIndexScale: 1e12 or 1e15
baseMinForRewards: 1_000e18 (is this too high? similar to USDC market, this would be around $1mn)
```

**Review by kevjin — APPROVED** [review]
